src/AstriaOracle.sol

```diff
@@ -36,21 +46,37 @@ contract AstriaOracle {
     // occurs when attempting to update the price for an uninitialized currency pair
     error UninitializedCurrencyPair(uint256 index);
 
+    // occurs when attempting to initialize an unauthorized currency pair, when authorization is required
+    error UnauthorizedCurrencyPair(bytes32 currencyPair);
+
     modifier onlyOracle() {
         require(msg.sender == ORACLE, "AstriaOracle: only oracle can update");
         _;
     }
 
-    constructor(address _oracle) {
+    constructor(address _oracle, bool _requireCurrencyPairAuthorization) Ownable(msg.sender) {
         ORACLE = _oracle;
+        requireCurrencyPairAuthorization = _requireCurrencyPairAuthorization;
+    }
+
+    function setRequireCurrencyPairAuthorization(bool _requireCurrencyPairAuthorization) external onlyOwner {
+        requireCurrencyPairAuthorization = _requireCurrencyPairAuthorization;
+    }
+
+    function authorizeCurrencyPair(bytes32 _currencyPair) external onlyOwner {
+        authorizedCurrencyPairs[_currencyPair] = true;
     }
 
     function initializeCurrencyPair(bytes32 _currencyPair, uint8 _decimals) external onlyOracle {
+        if (requireCurrencyPairAuthorization && !authorizedCurrencyPairs[_currencyPair]) {
+            revert UnauthorizedCurrencyPair(_currencyPair);
+        }
+
         currencyPairInfo[_currencyPair] = CurrencyPairInfo(true, _decimals);
         emit CurrencyPairInitialized(_currencyPair, _decimals);
     }
 
-    function updatePriceData(bytes32[] memory _currencyPairs, uint128[] memory _prices) external onlyOracle {
+    function setPrices(bytes32[] memory _currencyPairs, uint128[] memory _prices) external onlyOracle {
```

Comment:
This already seems to be the case, but we should maybe document somewhere on how authorization works.

Authorization is only for a currencyPair to be initialized, once initialized it always exists.